### PR TITLE
fix(1082): [6] add regex validation of commit branch filter.

### PR DESCRIPTION
**Pull request by wdstar**
## Context

SCM Branch-specific jobs

## Objective

This PR adds  regex validation of commit branch filter (e.g. `/^user-.*$/`).

## References

- [Design PR](https://github.com/screwdriver-cd/screwdriver/pull/1092)
- https://github.com/screwdriver-cd/screwdriver/issues/1082

### CODE REVIEW

File: config/regex.js [hunk 1]
```diff
@@ -49,6 +49,8 @@ module.exports = {
     // Can be ~pr, ~commit, or ~commit:branchName, or ~sd@123:component
     // Note: if you modify this regex, you must modify `sdJoi` definition in the `config/job.js`
     TRIGGER: /^~(sd@\d+:[\w-]+|pr|commit(:(.+))?)$/,
+    // Can be ~commit or ~commit:master
```
Comment (yokawara) [inline config/regex.js]: `~commit:branchName` is better.

Comment (yuichi10) [reply config/regex.js]: Thanks I will fix!

File: config/job.js [hunk 2]
```diff
@@ -4,6 +4,41 @@ const Annotations = require('./annotations');
...
+    name: 'string',
+    language: {
+        commitBranch: 'needs to be a commit branch string/regex'
+    },
+    rules: [
+        {
+            name: 'commitBranch',
+            validate(params, value, state, options) {
+                const matched = Regex.TRIGGER.exec(value);
+                const brFilter = matched[3];
```
Comment (tk3fftk) [inline config/job.js]: It seems to be a magic number.

File: config/job.js [hunk 3]
```diff
@@ -4,6 +4,42 @@ const Annotations = require('./annotations');
...
+    language: {
+        commitBranch: 'needs to be a commit branch string/regex'
+    },
+    rules: [
+        {
+            name: 'commitBranch',
+            validate(params, value, state, options) {
+                const regexPos = 3;
+                const matched = Regex.TRIGGER.exec(value);
+                const brFilter = matched[regexPos];
```
Comment (tk3fftk) [inline config/job.js]: I think we should also add comments to these parsing.

Comment (yuichi10) [reply config/job.js]: Sorry my variable name is not good I will fix the name regexPos to specificBranchPos.
Then, it give you sense which part this code try to get. 
After this correction and see the Regex.TRIGGER, maybe you can understand easily.

What do you think?

Comment (yuichi10) [reply config/job.js]: I added comments.

File: config/job.js [hunk 4]
```diff
@@ -4,6 +4,44 @@ const Annotations = require('./annotations');
...
+    base: joi.string(),
+    name: 'string',
+    language: {
+        commitBranch: 'needs to be a commit branch string/regex'
+    },
+    rules: [
+        {
+            name: 'commitBranch',
+            validate(params, value, state, options) {
+                const specificBranchPos = 3;
```
Comment (tkyi) [inline config/job.js]: I think it's more clear this is a constant, hardcoded value if you put it in all caps like https://github.com/screwdriver-cd/executor-k8s-vm/blob/master/index.js#L14

Comment (yuichi10) [reply config/job.js]: I fixed!

**Pull request merged**